Doc/howto/regex.rst

```diff
@@ -850,20 +887,29 @@ subgroups, from 1 up to however many there are. ::
    >>> m.groups()
    ('abc', 'b')
 
-Backreferences in a pattern allow you to specify that the contents of an earlier
-capturing group must also be found at the current location in the string.  For
-example, ``\1`` will succeed if the exact contents of group 1 can be found at
-the current position, and fails otherwise.  Remember that Python's string
-literals also use a backslash followed by numbers to allow including arbitrary
-characters in a string, so be sure to use a raw string when incorporating
-backreferences in a RE.
+Backreferences in a pattern allow you to specify that the contents of an
+earlier capturing group must also be found at the current location in the
+string. For example, ``\2`` will reference the substring matched by group 2,
+succeeding only if those exact contents are found at the current position
+within the string.
+
+(Remember that Python's string literals also use a backslash followed by
+numbers for including arbitrary characters in a string, so be sure to use a
+raw string when incorporating backreferences in a RE.)
 
 For example, the following RE detects doubled words in a string. ::
 
-   >>> p = re.compile(r'\b(\w+)\s+\1\b')
+   >>> p = re.compile(r'\b(\w+)\b\s+\1\b')
```

Comment:
The second `\b` was removed intentionally. It is not needed here.

It is worth also to use possessive qualifiers here.

Reply:
But it's fine to keep the second `\b`, and when modifying the example for some other context it might be useful. So I'd be fine with keeping it too. (Note that it's mentioned in the text below also.)

(Also, what's a possessive qualifier?)

Reply:
Not exactly this example, but see the conversation in https://github.com/python/cpython/pull/21420 about redundant `\b`.

This example was fixed in https://github.com/python/cpython/pull/4443. It was incorrect without `\b` at the end, but `\b` between `\w` and `\s` is redundant by definition.

Sorry, not "possessive qualifier" but  "possessive quantifier" (although in some documents they are named "qualifiers"). A possessive quantifier is a quantifier without backtracking. It is written by adding `+` to the quantifier (as non-greed quantifiers are written by adding `?`). For example, when try to match the pattern with greedy quantifiers `\b(\w+)\s+\1\b` in "then the", a dumb backtracking engine will try to match "then then", fail, backtrack and try to match consequentially "the ", "th ", "t " until it give up. But with possessive quantifier `\b(\w++)\s++\1\b` it will not backtrack and fail quicker. It is a new feature in Python 3.11. Even if it is supported in most modern RE engines, it is relatively little known, because it was not initially supported in old RE engines.

See https://www.regular-expressions.info/possessive.html

Reply:
OK, I've removed the second \b and edited the text below a bit.